File: CodingSamples/DotNet/Advanced/Microservices/RESTTest/RESTTest/ServerApp/Resources/SalesAgentApi.cs
using System.IdentityModel.Tokens.Jwt;
using ServerApp.Security;

namespace ServerApp.Resources;

public class SalesAgentApi
{
    public static async Task<IResult> SignIn(string id, int passcode)
    {
        string agentMailId = id + "@sales.met.edu"; 
        if(passcode == 9820)
        {
            await OtpHelper.MailPasscodeAsync(agentMailId, "admin@sales.met.edu");
            return Results.Ok();
        }
        if(OtpHelper.VerifyPasscode(agentMailId, passcode))
        {
            var token = JwtHelper.CreateToken(id);
            return Results.Text(token);
        }
        return Results.Unauthorized();
    }
}

File: CodingSamples/DotNet/Advanced/Microservices/RESTTest/server/src/main/java/app/resources/SalesAgentApi.java
package app.resources;

import app.security.JwtHelper;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.PathParam;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.core.Response.Status;

@Path("/api/sales")
public class SalesAgentApi {
    
    @GET
    @Path("/{id}/{code}")
    public Response signIn(@PathParam("id") String agentId, @PathParam("code") int passcode) {
        if(passcode == 9820){
            String token = JwtHelper.createToken(agentId);
            return Response.ok(token).build();
        }
        return Response.status(Status.UNAUTHORIZED).build();
    }
}
